core/src/test/scala/org/apache/spark/sql/insertion/BatchWritePKAndIndexSuite.scala

```diff
@@ -116,6 +117,9 @@ class BatchWritePKAndIndexSuite
   // https://github.com/pingcap/tispark/issues/2452
   test("test duplicate unique indexes are not deleted error") {
     tidbStmt.execute("drop table if exists `tispark_test`.`t`")
+    if (StoreVersion.minTiKVVersion("5.0.0", this.ti.tiSession.getPDClient)) {
```

Comment:
Isn't the logic totally reversed? You need to skip the test if it's smaller than version 5.0.0, but you are skipping it when it's bigger than version 5.0.0.